io/src/main/java/org/commonjava/maven/ext/io/PomIO.java

```diff
@@ -248,49 +257,63 @@ private void write( final Project project, final File pom, final Model model )
+
+            // Annoyingly the document is private but we need to access it in order to ensure the model is written to the Document.
+            //
+            // Currently the fields we want to access are private - https://issues.apache.org/jira/browse/MRELEASE-1044 requests
+            // them to be protected to avoid this reflection.
+            Document doc = (Document) FieldUtils.getDeclaredField( JDomModelETL.class, "document", true ).get( etl );
+
+            jdomModelConverter.convertModelToJDOM( model, doc );
+
+            if ( project.isExecutionRoot() )
             {
-                @Override
-                public void postProcess( final Document doc )
+                // Previously it was possible to add a comment outside of the root element (which maven3-model-jdom-support handled)
+                // but the release plugin code only takes account of code within the root element and everything else is handled separately.
+                //
+                String outtro = (String) FieldUtils.getDeclaredField( JDomModelETL.class, "outtro", true ).get( etl );
+
+                String commentStart = ls.value() +
+                                "<!--" +
+                                ls.value();
+                String commentEnd = ls.value() +
+                                "-->" +
+                                ls.value();
+
+                if ( outtro.equals( ls.value() ) )
                 {
-                    // Only add the modified by to the top level pom.
-                    if ( project.isExecutionRoot() )
-                    {
-                        final Iterator<Content> it = doc.getContent( new ContentFilter( ContentFilter.COMMENT ) )
-                                                        .iterator();
-                        while ( it.hasNext() )
-                        {
-                            final Comment c = (Comment) it.next();
+                    logger.debug( "Outtro contains newlines only" );
 
-                            if ( c.toString().contains( MODIFIED_BY ) )
-                            {
-                                it.remove();
-                            }
-                        }
-
-                        Comment comment = new Comment( mjw.getLineSeparator()
-                                + "Modified by POM Manipulation Extension for Maven " + manifestInformation
-                                + mjw.getLineSeparator() );
-                        doc.addContent( Collections.<Content>singletonList( comment ) );
-                    }
+                    outtro = commentStart + manifestComment + commentEnd;
                 }
-            });
-        }
-        catch ( final IOException e )
-        {
-            throw new ManipulationException( "Failed to read POM for rewrite: {}. Reason: ", pom, e.getMessage(), e );
+                else
+                {
+                    outtro = outtro.replaceAll( "Modified by.*", manifestComment );
```

Comment:
Probably this could use the `MODIFIED_BY + .*` constant that you already have defined.

Reply:
I don't think so - the modified by includes a version and we want to ensure we replace it by the correct version. Its feasible it could have an older version.